src/components/data-management/DownloadData.jsx

```diff
@@ -0,0 +1,171 @@
+/* eslint-disable import/no-unresolved */
+import React, { useEffect } from 'react';
+import _ from 'lodash';
+import {
+  Menu, Tooltip, Dropdown, Button,
+} from 'antd';
+
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { saveAs } from 'file-saver';
+import downloadTypes from 'utils/data-management/downloadTypes';
+import { getFromApiExpectOK } from 'utils/getDataExpectOK';
+import pushNotificationMessage from 'utils/pushNotificationMessage';
+import endUserMessages from 'utils/endUserMessages';
+import pipelineStatus from '../../utils/pipelineStatusValues';
+import { exportQCParameters, filterQCParameters } from '../../utils/data-management/exportQCParameters';
+import { loadBackendStatus } from '../../redux/actions/backendStatus/index';
+
+const DownloadData = (props) => {
+  const {
+    activeProjectUuid,
+  } = props;
+  const dispatch = useDispatch();
+  const activeProject = useSelector((state) => state.projects[activeProjectUuid]);
+  const experimentSettings = useSelector((state) => state.experimentSettings);
+  const backendStatus = useSelector((state) => state.backendStatus);
+  const samples = useSelector((state) => state.samples);
+  const projects = useSelector((state) => state.projects);
+
+  // Change if we have more than one experiment per project
+  const experimentId = activeProject?.experiments[0];
+
+  useEffect(() => {
+    if (experimentId && !backendStatus[experimentId]) {
+      dispatch(loadBackendStatus(experimentId));
+    }
+  }, [experimentId]);
+
+  const getQcHasRun = () => (
+    experimentId
+    && (backendStatus[experimentId]?.status.pipeline?.status === pipelineStatus.SUCCEEDED)
+  );
+
+  const getGem2sHasRun = () => (
+    experimentId
+    && (backendStatus[experimentId]?.status?.gem2s?.status === pipelineStatus.SUCCEEDED)
+  );
+
+  const getAllSamplesAnalysed = () => {
+    // Returns true only if there is at least one sample in the currently active
+    // project AND all samples in the project have been analysed.
+    if (!activeProject?.samples?.length) {
+      return false;
+    }
+    const steps = Object.values(_.omit(experimentSettings?.processing, ['meta']));
+
+    return steps.length > 0
+      // eslint-disable-next-line no-prototype-builtins
+      && activeProject?.samples?.every((s) => steps[0].hasOwnProperty(s));
+  };
+
+  const gem2sHasRun = getGem2sHasRun();
+  const qcHasRun = getQcHasRun();
+  const allSamplesAnalysed = getAllSamplesAnalysed();
```

Comment:
This would be no longer necessary as all of them are going to be state variables.